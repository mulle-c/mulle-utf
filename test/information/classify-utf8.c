#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>

static int   test( mulle_utf8_t *text, size_t len)
{
   struct mulle_utf_information   info;

   if( mulle_utf8_information( text, len, &info))
   {
      printf( "Offending character at position %ld (%u)\n",
                           (long) ((mulle_utf8_t *) info.invalid - text),
                           *(mulle_utf8_t *) info.invalid);
      return( 1);
   }
   return( 0);
}


int  main()
{
   int   fails;

   fails  = 0;
   fails += test( (mulle_utf8_t []) { 'I', 't', 226, 's' }, 4);
   return( 0);
}

