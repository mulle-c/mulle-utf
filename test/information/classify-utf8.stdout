Offending character at position 2 (226)
